### DISCUSSION TITLE
How to Use Machine ID to generate auth token for slack plugin

### DISCUSSION THREAD

**Discussion by SriNandan33** [gravitational/teleport#28995] [Q&A] [answered]
We followed [slack plugin installation docs](https://goteleport.com/docs/access-controls/access-request-plugins/ssh-approval-slack/?scope=cloud) and set up the slack plugin using **Helm Chart**.  During the development we created the credentials using the following command `tctl auth sign --user=access-plugin --out=auth.pem` and manually created a K8s secret out of it. It worked perfectly fine.

However, it is recommended to use MachineID to generate credentials for plugins in production. We couldn't find any guides around it. How can we use Machine ID to generate credentials for the slack plugin, and create a K8s secret with the generated credentials in the same K8s cluster where we are running our Teleport kube agent, Slack plugin helm charts? Any help is appreciated, Thanks!

**Comment by webvictim** [comment, marked as answer]
This isn't something that's officially supported at the moment - it's on our radar to fix as part of https://github.com/gravitational/teleport/issues/29048

The issue with doing this manually at the moment is that by default, Machine ID generates certs that are valid for 1 hour (and renewed every 20 minutes), but the Teleport plugins (Slack et al) can't currently watch and re-read certificates on disk when they change. As such you'd have to restart the Slack plugin container at least once an hour to re-read the updated certs.

@strideynet may have more context I'm missing, but know that we're working on a solution to this and will definitely make it possible to use Machine ID credentials in future.
